### PR TITLE
Devtools gen:asset Enhancements

### PR DESCRIPTION
**Pull request by lexfm**
### Change Type (required)
Creation and generation of:

- Storybook starter story setup 
- Basic flow to showcase in our starter story for Storybook
- .mdx document for storybook
- .mdx docsite documentation 

Links to the necessary files:

- docs/storybook/.storybook/preview.ts: For linking the asset plugin to be used to render the story. 
- .bazelignore: To list the asset node_modules directory to be ignored by the build process
- pnpm-workspace.yaml:  To list the asset package as part of the Devtools repository
- plugin/README.md: To list asset for React Player instance in document.
- plugin/src/index: To list the asset Type and component DSL representation export for the plugin registry.
- plugin/src/plugins/AssetsRegistryPlugin.tsx: To list the asset to register to the  React Player instance.
- plugin/src/plugins/TransformPlugin.ts: To map the asset transform function to its asset when the plugin is registered and loaded.
- .mdx document linking and navigation: for adding the newly created mdx for docs site to its sidebar navigation.
- plugin/package.json: Listing new generated asset as workspace dependency for the assets plugin.
- plugin/BUILD: Listing new generated asset package node_modules on bazel plugin deps.
- plugin/README: Listing new generated asset on README file list.

[Full proposal](https://docs.google.com/document/d/1cuRUxPrksMKXOkzidLC7sHimf691XyAZbRgPyWBKWUA/edit)

- [ ] `patch`
- [ ] `minor`
- [ ] `major`

### CODE REVIEW

File: asset-template/src/component/index.tsx.hbs [hunk 1]
```diff
@@ -14,7 +14,7 @@ const use{{pascalCase assetName}}Pros = (props: Transformed{{pascalCase assetNam
 };
 
 export const {{pascalCase assetName}}Component = (props: Transformed{{pascalCase assetName}}) => {
-  const { children, ...buttonProps } = use{{pascalCase assetName}}Pros(props);
+  const { children, ...buttonProps } = use{{pascalCase assetName}}Props(props);
```
Comment (rafbcampos) [inline asset-template/src/component/index.tsx.hbs:17]: Let's use a generic name like `...rest`, for example. The `buttonProps` was kept by mistake on the former generator.

**Review by rafbcampos — APPROVED** [review]
This is an excellent improvement over the former generator. There is a minor comment on the `buttonProps`; other than that, it is good to merge.

**Review by mercillo — APPROVED** [review]

**Comment by mercillo** [comment]
just curious too on why all the file names got changed with .hbs

**Comment by lexfm** [comment]
> just curious too on why all the file names got changed with .hbs

`.hbs` is the handlebars extension that is supported by `plopjs` as source or template for dynamically creating new files.

**Pull request merged**